src/uv_tcp.c

```diff
@@ -97,3 +99,22 @@ void raft_uv_tcp_close(struct raft_uv_transport *transport)
     struct UvTcp *t = transport->impl;
     raft_free(t);
 }
+
+int raft_uv_tcp_set_bind_address(struct raft_uv_transport *transport,
+                                 const char *address)
+{
+    struct UvTcp *t = transport->impl;
+    struct sockaddr_in addr;
+    int rv;
+
+    rv = uvIpParse(address, &addr);
+    if (rv != 0) {
+        return RAFT_INVALID;
+    }
+
+    t->bind_address = strdup(address);
```

Comment:
This memory is never freed, we also usually use the allocator functions from `src/heap.c`, you could do something similar as here https://github.com/canonical/raft/blob/3769e75e5adbb6954c557403818cb53cc9aa14e1/src/raft.c#L44